Add `Clear` method to `BaseAppender`

EtgarDev opened the pull request:
## Summary
This Pull Request adds a new `Clear` function to the `Appender`, enabling the clearing of pending data without flushing. 

### Motivation
When closing an `Appender`, a `Flush` will also be triggered.
However, sometimes, we wouldn't want to flush the data, for instance, when there is a problem with the flushing.
In such case, we would sometimes prefer to only clear the appender and close it.
We enable this capability by exposing the `Clear` method.

### Changes
- `Clear` method
- Added a test for verifying the `Clear` method correctly clears data and that the next `Flush` won't write the data that should have been cleared.
- Added a test that simulates a Heavy `QueryAppender` that should be interrupted and then cleared, before being closed, to ensure the `Clear` method actually "enables" the `Close` method for such scenarios.

### Upcoming Next
A PR for exposing the `Clear` method through the C-API.

EtgarDev committed:
Add `Clear` method to `BaseAppender` + a test case

taniabogatsch commented on test/appender/test_appender.cpp:
```diff
@@ -777,3 +779,77 @@ TEST_CASE("Test appending rows with an active column list", "[appender]") {
+	    app.Close();
+	});
+
+	auto status = future.wait_for(std::chrono::seconds(1));
```
Maybe we can even lower the time here - if we cleared the data, `close` should be very swift!

taniabogatsch reviewed (changes requested):
Hi @EtgarDev ! Thanks for the PR, looking good already! I just left a few comments. :)

EtgarDev commented on test/appender/test_appender.cpp:
```diff
@@ -777,3 +779,77 @@ TEST_CASE("Test appending rows with an active column list", "[appender]") {
+		try {
+			app.Flush();
+		} catch (...) {
+			// Expected when the query is interrupted
```
Yep I'll do it ;)

EtgarDev committed:
Update `Appender::Clear()` test case to cover also resetting the chunk after at least 1 flush

EtgarDev committed:
Adjust `Appender::Close()` test to reduce async wait time to 50ms

EtgarDev committed:
Enhance `Appender::Flush()` test to validate interrupt exception type

taniabogatsch reviewed (commented):
Thanks for the quick changes, left two more comments. :)

EtgarDev committed:
Simplify appender test, ensuring proper collection clearing validation

EtgarDev commented on test/appender/test_appender.cpp:
```diff
@@ -777,3 +779,89 @@ TEST_CASE("Test appending rows with an active column list", "[appender]") {
+	Appender appender(con, "main", "ints");
+
+	// Those should be flushed due to the max chunk size trigger
+	for (idx_t i = 0; i < BaseAppender::DEFAULT_FLUSH_COUNT; i++) {
```
OK we spoke about it and understood that for some reason I thought you referred to the maximum rows in an appender. Fixed :)

EtgarDev committed:
Adjust appender test comment formatting

krleonid committed:
Add duckdb_appender_clear C API function

- Added duckdb_appender_clear to appender JSON with detailed description
- Added implementation in appender-c.cpp that calls BaseAppender::Clear()
- Added duckdb_appender_clear to v1.2.0 API struct
- Added comprehensive test in test_capi_appender.cpp
- Regenerated headers using make generate-files

EtgarDev committed:
Refine appender test to use fixed `rows_to_append` for consistent behavior

EtgarDev committed:
Align appender test with `DEFAULT_FLUSH_COUNT` to support all vector_sizes

EtgarDev committed:
temporarily disabling the test when the vector size is not the default vector size

EtgarDev committed:
- Fixed test of CAPI appender to support the case of very small vector size configuration.
- Remove the "skipping" workaround from another test
